src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/services/citizen/SettlementReachedCitizenNotificationServiceTest.java

```diff
@@ -0,0 +1,174 @@
+        emailData.put(FRONTEND_BASE_URL_KEY, FRONTEND_BASE_URL);
+
+        verifyEmailData(emailDataArgument.getAllValues().get(4), emailData);
+        assertThat(senderArgument.getAllValues().get(4)).isEqualTo(claim.getSubmitterEmail());
+    }
+
+    @Test
+    public void verifyNotificationsWhenCounterSignedByClaimant() throws Exception {
+        given(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .willReturn(PDF_CONTENT);
+
+        claimantRequestFor("reject");
+        prepareClaimantOffer();
+        defendantRequestFor("accept");
+        claimantRequestFor("countersign");
+
+        verify(notificationClient, atLeast(10))
+            .sendEmail(templateArgument.capture(), senderArgument.capture(), emailDataArgument.capture(),
+                referenceArgument.capture());
+
+        verifyConfirmationToClaimantOnClaimantCounterSign();
+        verifyNotificationToDefendantOnClaimantCounterSign();
+    }
+
+    private void verifyConfirmationToClaimantOnClaimantCounterSign() {
+        assertThat(templateArgument.getValue()).isEqualTo(OFFER_COUNTER_SIGNED_EMAIL_BY_ORIGINATOR);
+
+        assertThat(referenceArgument.getValue())
+            .matches("to-claimant-agreement-counter-signed-by-claimant-notification-"
+                + CLAIM_REFERENCE_NUMBER);
+
+        Map<String, String> emailData = new HashMap<>();
+        emailData.put(COUNTER_SIGNING_PARTY_KEY, claim.getClaimData().getClaimant().getName());
+        emailData.put(FRONTEND_BASE_URL_KEY, FRONTEND_BASE_URL);
+
+        verifyEmailData(emailDataArgument.getValue(), emailData);
+        assertThat(senderArgument.getValue()).isEqualTo(claim.getSubmitterEmail());
+    }
+
+    private void verifyEmailData(Map<String, String> output, Map<String, String> expected) {
+        output.entrySet().stream()
+            .filter(e -> !e.getKey().equals("claimReferenceNumber"))
+            .forEach(e -> assertThat(output.get(e.getKey())).isEqualTo(expected.get(e.getKey())));
+
+        assertThat(output.get("claimReferenceNumber")).matches(CLAIM_REFERENCE_PATTERN);
+
+    }
+
+    private void verifyNotificationToDefendantOnClaimantCounterSign() {
+        assertThat(templateArgument.getAllValues().get(8)).isEqualTo(OFFER_COUNTERSIGNED_EMAIL_BY_OTHER_PARTY);
+
+        assertThat(referenceArgument.getAllValues().get(8))
+            .matches("to-defendant-agreement-counter-signed-by-claimant-notification-"
+                + CLAIM_REFERENCE_NUMBER
+            );
+
+        Map<String, String> emailData = new HashMap<>();
+        emailData.put(COUNTER_SIGNING_PARTY_KEY, claim.getClaimData().getClaimant().getName());
+        emailData.put(FRONTEND_BASE_URL_KEY, FRONTEND_BASE_URL);
+
+        verifyEmailData(emailDataArgument.getAllValues().get(8), emailData);
+
+        assertThat(senderArgument.getAllValues().get(8))
+            .isEqualTo(claim.getClaimData().getDefendant().getEmail().orElse(null));
```

Comment:
The same here, Defendant email is taken from claim directly, not from `claimData`